react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useEffect } from 'react';
+
+function Carousel(props) {
+  const [isClicked, setIsClicked] = useState(0);
+  const Arraylength = props.images.length;
+
+  const onClickFnIcon = index => {
+    setIsClicked(prestate => {
+      clearInterval(intervalID);
+      carousal();
+      return index;
+    });
+  };
+  const icons = props.images.map((icon, index) => {
+    return <i key={index} className={isClicked === index ? 'fa-solid fa-circle icon-adjust' : 'fa-regular fa-circle icon-adjust'}
+    onClick={() => { onClickFnIcon(index); }}></i>;
+  });
+
+  let intervalID;
+  const carousal = () => {
+    intervalID = setInterval(() => {
+      setIsClicked(prestate => {
+        prestate++;
+        if (prestate === Arraylength) {
+          prestate = 0;
+        }
+        return prestate;
+      });
+    }, 2000);
+  };
+
+  useEffect(() => {
+    carousal();
+    return () => clearInterval(intervalID);
+  });
+  const CarouPics = ({ index, isClicked, setIsClicked, data }) => {
+    return (
+      <div key={'container' + index}>
+        <img key={'image' + index} src={data} className={isClicked === index ? 'image' : 'image hidden'} />
+      </div>
+    );
+  };
```

Comment:
These CarouPics, LeftChevron, and RightChevron components should be defined outside of the Carousel component. In React apps, we don't define components inside of each other.